.github/workflows/tests.yaml

```diff
@@ -0,0 +1,104 @@
+name: "Tests"
+
+concurrency:
+  group: "tests-${{ github.head_ref || github.run_id }}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  tests:
+    name: "PHP ${{ matrix.php }} | PostgreSQL ${{ matrix.postgresql }}"
+    runs-on: "ubuntu-latest"
+    container:
+      image: "php:${{ matrix.php-version }}-cli-alpine"
+      env:
+        DATABASE_URL: "postgresql://main:main@postgresql:5432/main?serverVersion=${{ matrix.postgresql }}&charset=utf8"
+    services:
+      postgresql:
+        image: "postgres:${{ matrix.postgresql-version }}-alpine"
+        env:
+          POSTGRES_USER: main
+          POSTGRES_PASSWORD: main
+          POSTGRES_DB: main
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php:
+          - "7.4"
+        postgresql:
+          - "11"
+        include:
+          - php: "7.4"
+            php-version: "7.4.1"
+          - postgresql: "11"
+            postgresql-version: "11.7"
+
+    steps:
+      - name: "Install OS dependencies"
+        run: "apk add --no-cache bash git icu-dev libzip-dev unzip zip"
+
+      - name: "Adjust allowed PHP memory"
+        run: echo 'memory_limit = -1' > $PHP_INI_DIR/conf.d/memory-limit.ini;
+
+      - name: "Install PHP extensions"
+        shell: bash
+        run: |
+          wget https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions --quiet -O /usr/local/bin/install-php-extensions
+          chmod +x /usr/local/bin/install-php-extensions
+          install-php-extensions intl pdo_pgsql zip
+
+      - name: "Install composer"
+        run: curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin --filename=composer --version=2.6.6
+
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+        with:
+          ref: ${{ github.event.pull_request.head.ref || '' }}
```

Comment:
we are in a detached HEAD mode in pull request, so we want to move back to the actual PR branch - without this package synchronization tests are failing :angry: